torch/utils/_stats.py

```diff
@@ -1,20 +1,19 @@
-# mypy: allow-untyped-defs
 # NOTE! PLEASE KEEP THIS FILE *FREE* OF TORCH DEPS! IT SHOULD BE IMPORTABLE ANYWHERE.
 # IF YOU FEEL AN OVERWHELMING URGE TO ADD A TORCH DEP, MAKE A TRAMPOLINE FILE A LA torch._dynamo.utils
 # AND SCRUB AWAY TORCH NOTIONS THERE.
 import collections
 import functools
-from typing import OrderedDict
+from typing import Any, Callable, OrderedDict
 
 simple_call_counter: OrderedDict[str, int] = collections.OrderedDict()
 
-def count_label(label):
+def count_label(label: str) -> None:
     prev = simple_call_counter.setdefault(label, 0)
     simple_call_counter[label] = prev + 1
 
-def count(fn):
```

Comment:
This looks like a decorator. You can't just decorate a decorator like this. The problem is that this will scrub away the annotations on the decorated item.

For example:
```
@count
def foo(a: int) -> str:
   ...
```
will appear to be of type `Callable[..., Any]` because that's what the decorator returns.

Instead you need to use ParamSpec and TypeVar to correctly pass the input type to the output type:
```
_P = ParamSpec("_P")
_T = TypeVar("_T")

def count(fn: Callable[_P, _T]) -> Callable[_P, _T]:
```